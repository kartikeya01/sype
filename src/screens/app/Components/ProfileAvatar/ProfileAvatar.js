import React from 'react';
import {View, StyleSheet} from 'react-native';
import {Avatar, Title} from 'react-native-paper';
export default class ProfileAvatar extends React.Component {
  constructor(props) {
    super(props);
  }

  handleCommentClick = () => {};

  render() {
    return (
      <View style={styles.rowContainer}>
        <Avatar.Image size={24} source={{uri: this.props.img}} />
        <Title style={styles.titleStyle}>{this.props.username}</Title>
      </View>
    );
  }
}
const styles = StyleSheet.create({
  titleStyle: {
    marginHorizontal: 10,
  },
  rowContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});
