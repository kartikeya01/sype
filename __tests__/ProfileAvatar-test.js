import 'react-native';
import React from 'react';
import ProfileAvatar from '@screens/app/Components/ProfileAvatar/ProfileAvatar';
import checkPropTypes from 'check-prop-types';
var assert = require('assert');
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

test('profile avarar snapshot test', () => {
  const treeWithoutProps = renderer.create(<ProfileAvatar />).toJSON();
  expect(treeWithoutProps).toMatchSnapshot();
  const treeWithUsername = renderer.create(<ProfileAvatar username="emanasia" />).toJSON();
  expect(treeWithUsername).toMatchSnapshot();
  const treeWithUsernameAndImage = renderer.create(<ProfileAvatar username="emanasia" img="https://images.app.goo.gl/HPsoQwfF6dpaqUqG8" />).toJSON();
  expect(treeWithUsernameAndImage).toMatchSnapshot();
});

function testPropTypesWithoutErrors(username) {
  it(
    'testing ProfileAvatar Creation with proper props where ' +
      'username : ' +
      username , () => {
      const wrapper = shallow(
        <ProfileAvatar username={username} />,
      );
      const instance = wrapper.instance();
      const usernameProp = instance.props.username;
      expect(username).toEqual(usernameProp);
    },
  );
}

// test numbers, arrays, strings, booleans combinations of other data types
describe('testing ProfileAvatar values, should not throw errors', () => {
  testPropTypesWithoutErrors('e805');
  testPropTypesWithoutErrors('bsefa');

});

function testPropTypesWithErrors(username, errorMsg, testDescription) {
  it('Throws failed propType Error for ' + testDescription, () => {
    let result = checkPropTypes(
      ProfileAvatar.propTypes,
      {username: username},
      'prop',
      ProfileAvatar.username,
    );
    assert(result === errorMsg);
  });
}

describe('testing ProfileAvatar values, should throw errors', () => {
  testPropTypesWithErrors(
    null,
    'Failed prop type: The prop `username` is marked as required in `<<anonymous>>`, but its value is `null`.',
    'not providing username',
  );
  testPropTypesWithErrors(
    true,
    'Failed prop type: Invalid prop `username` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
    'providing username value of true and not string',
  );
  testPropTypesWithErrors(
    false,
    'Failed prop type: Invalid prop `username` of type `boolean` supplied to `<<anonymous>>`, expected `string`.',
    'providing username value of false and not string',
  );
  testPropTypesWithErrors(
    undefined,
    'Failed prop type: The prop `username` is marked as required in `<<anonymous>>`, but its value is `undefined`.',
    'not providing username',
  );
  testPropTypesWithErrors(
    234,
    'Failed prop type: Invalid prop `username` of type `number` supplied to `<<anonymous>>`, expected `string`.',
    'providing username value of number and not string',
  );
  testPropTypesWithErrors(
    234.5,
    'Failed prop type: Invalid prop `username` of type `number` supplied to `<<anonymous>>`, expected `string`.',
    'providing username value of number and not string',
  );

});
