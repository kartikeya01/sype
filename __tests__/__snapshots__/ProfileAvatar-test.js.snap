// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile avarar snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#6200ee",
          "borderRadius": 15,
          "height": 30,
          "width": 30,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 15,
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "marginHorizontal": 10,
            },
          ],
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            Object {
              "marginHorizontal": 10,
            },
          ],
        ],
      ]
    }
  />
</View>
`;
